remote-build-coordinator/src/main/java/org/jboss/pnc/remotecoordinator/builder/datastore/DatastoreAdapter.java

```diff
@@ -416,12 +426,38 @@ public BuildRecord storeResult(BuildTaskRef buildTask, Optional<BuildResult> bui
                     userLog.error(message);
                     break;
             }
+            errorLog.append(message);
         } else {
             buildRecordBuilder.status(SYSTEM_ERROR);
-            userLog.error("Build status: Missing Build Result!");
+            String message = "Build status: Missing Build Result!";
+            errorLog.append(message);
+            userLog.error(message);
         }
         log.debug("Storing ERROR result of BCA: " + buildTask.getIdRev().toString() + " to datastore.", e);
-        return datastore.storeCompletedBuild(buildRecordBuilder, Collections.emptyList(), Collections.emptyList());
+        BuildRecord buildRecord = datastore
+                .storeCompletedBuild(buildRecordBuilder, Collections.emptyList(), Collections.emptyList());
+        uploadLogs(errorLog.toString());
+        return buildRecord;
+    }
+
+    private void uploadLogs(String errorLog) {
+        try {
+            Map<String, String> headersFromMDC = MDCUtils.getHeadersFromMDC();
+
+            LogMetadata logMetadata = LogMetadata.builder()
+                    .tag("build-log")
+                    .endTime(OffsetDateTime.now())
+                    .loggerName("orch")
+                    .processContext(headersFromMDC.get(MDCHeaderKeys.PROCESS_CONTEXT))
+                    .processContextVariant(headersFromMDC.get(MDCHeaderKeys.PROCESS_CONTEXT_VARIANT))
+                    .tmp(headersFromMDC.get(MDCHeaderKeys.TMP))
+                    .requestContext(headersFromMDC.get(MDCHeaderKeys.REQUEST_CONTEXT))
+                    .build();
+            bifrostLogUploader.uploadString(errorLog, logMetadata);
+        } catch (BifrostUploadException ex) {
```

Comment:
is this the only Exception that the bifrost client throws?

Reply:
All the expected exception are under that one.